### PR TITLE
18967: UQHP Final Eligibility Notice (Renewal) - IVL_FRE

### PR DESCRIPTION
**Pull request by davidplappert**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [18967](https://devops.dchbx.org/redmine/issues/18967) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | 12/31/2017  |
| App-Dev Priority | UP-1 |
| Development Story Points | 2 |
| Peer Review Story Points | 0.5 |
| Ticket Author | @karashearer5 |

### CODE REVIEW

File: app/notices/ivl_notices/final_eligibility_notice_renewal_aqhp.rb [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    append_enrollment_information
+    if primary_member["aqhp_eligible"].upcase == "YES"
+      notice.tax_households = append_tax_household_information(primary_member)
+    end
+    notice.has_applied_for_assistance = check(primary_member["aqhp_eligible"])
+    notice.primary_firstname = primary_member["first_name"]
+  end
+
+  def append_member_information_for_aqhp(primary_member)
+    data.collect do |datum|
```
Comment (Battula) [inline app/notices/ivl_notices/final_eligibility_notice_renewal_aqhp.rb]: As per the template attached to the ticket, we don't most of this information.
All the logic in the notice template is based on the enrollment information.
Please remove any unnecessary data that you're trying to read from CSV.

File: script/ivl_renewal_notices.rb [hunk 2]
```diff
@@ -65,9 +65,9 @@ def valid_enrollments(person)
       #next if (members.any?{ |m| @excluded_list.include?(m["member_id"]) })
       primary_member = members.detect{ |m| m["dependent"].upcase == "NO"}
       next if primary_member.nil?
-      # next if (primary_member.present? && primary_member["policy.subscriber.person.is_dc_resident?"] == "FALSE")
+      next if (primary_member.present? && primary_member["policy.subscriber.person.is_dc_resident?"].upcase == "FALSE")
```
Comment (saikumar9) [inline script/ivl_renewal_notices.rb]: Excluded people who are not residents of **DC**

File: script/ivl_renewal_notices.rb [hunk 3]
```diff
@@ -65,9 +65,9 @@ def valid_enrollments(person)
...
       # next if members.select{ |m| m["policy.subscriber.person.is_incarcerated"] == "TRUE"}.present?
-      # next if (members.any?{ |m| (m["policy.subscriber.person.citizen_status"] == "non_native_not_lawfully_present_in_us") || (m["policy.subscriber.person.citizen_status"] == "not_lawfully_present_in_us")})
+      next if (members.any?{ |m| (m["policy.subscriber.person.citizen_status"] == "non_native_not_lawfully_present_in_us") || (m["policy.subscriber.person.citizen_status"] == "not_lawfully_present_in_us")})
```
Comment (saikumar9) [inline script/ivl_renewal_notices.rb]: People who are not lawfully present are excluded.

File: app/models/consumer_role.rb [hunk 4]
```diff
@@ -192,7 +192,7 @@ def all_types_verified?
   def is_type_outstanding?(type)
     case type
       when "DC Residency"
-        residency_denied? && !has_docs_for_type?(type)
+        residency_denied? && !has_docs_for_type?(type) && local_residency_outstanding?
```
Comment (saikumar9) [inline app/models/consumer_role.rb]: An additional check is added for **DC Residency** verification.

File: spec/models/consumer_role_spec.rb [hunk 5]
```diff
@@ -253,6 +253,20 @@
       end
     end
 
+    context "DC Residency" do
```
Comment (saikumar9) [inline spec/models/consumer_role_spec.rb]: Good test coverage for verifying DC Residency.

**Pull request merged**